### PR TITLE
refactor: pass reduceop into global_load

### PR DESCRIPTION
**Pull request by chaosagent**
prereq refactor for having more than one reduce in Linearizer (parallel, series).

### CODE REVIEW

File: tinygrad/codegen/linearizer.py [hunk 1]
```diff
@@ -95,10 +96,11 @@ def global_load(self, i:int, idxs:List[Node], acc=None, barrier:Optional[UOp]=No
     invalid_value = 0
     for idx, valid, rep_idx in zip(e_idxs, e_valids, iter_idxs(expand_vars)):
       this_const, idx, valid = (invalid_value, NumNode(0), NumNode(1)) if valid.max == 0 else (const, idx, valid)
-      key = f"{acc}{localtype}{'CONST'+str(this_const) if this_const is not None and acc is None else (buf.idx if isinstance(buf, MemBuffer) else cast(LocalBuffer, buf).name)}{idx.render()}{valid.render()}"  # noqa: E501
+      key = f"{acc is not None}{localtype}{'CONST'+str(this_const) if this_const is not None and acc is None else (buf.idx if isinstance(buf, MemBuffer) else cast(LocalBuffer, buf).name)}{idx.render()}{valid.render()}"  # noqa: E501
```
Comment (Qazalin) [inline tinygrad/codegen/linearizer.py:99]: I need to think about this line, changing this key is tricky.
Can you write a test for it? I have one example in `test_load_cache_const_bufs`

File: tinygrad/codegen/linearizer.py [hunk 2]
```diff
@@ -95,10 +96,11 @@ def global_load(self, i:int, idxs:List[Node], acc=None, barrier:Optional[UOp]=No
...
       if key not in self.load_cache:
         if acc is not None:
-          self.load_cache[key] = self.uops.add(UOps.DEFINE_ACC, localtype, (), dtypes.as_const(this_const, localtype), False, insert_before)
+          acc_const = dtypes.as_const(self.get_reduce_acc(acc), localtype)
```
Comment (Qazalin) [inline tinygrad/codegen/linearizer.py:102]: I think get_reduce_acc is already the correct dtype

File: tinygrad/codegen/linearizer.py (hunk 1 above)
Comment (Qazalin) [reply tinygrad/codegen/linearizer.py:99]: if you have multiple accs how are you ensuring they uop different DEFINE_ACCs?

File: tinygrad/codegen/linearizer.py (hunk 2 above)
Comment (Qazalin) [reply tinygrad/codegen/linearizer.py:102]: yeah, CI was ok with removing it

File: tinygrad/codegen/linearizer.py (hunk 1 above)
Comment (chaosagent) [reply tinygrad/codegen/linearizer.py:99]: each reduceop is associated with a disjoint set of output buffers. self.reduceops.index(acc) is a good idea.

Comment (chaosagent) [reply tinygrad/codegen/linearizer.py:99]: well, we don't have self.reduceops yet, and self.reduceops might end up being 2 dimensional for both parallel and sequence.

Comment (0xtimmy) [reply tinygrad/codegen/linearizer.py:99]: I figured it was fine to use the same local buffer for sequential reduceops as long as they're the same shape, that obviously won't work for parallel reduceops lol

there was some weirdness with the key, I also think we should pass `reduceop` to `global_load` and I was playing with `...{self.reduceops.index(reduceop) if reduceop else ''}...`

but I also start each reduce with an empty `self.load_cache` so it didn't end up needing to be changed

**Comment by Qazalin** [comment]
@chaosagent I'll merge if you either add a TODO or ideally a test for the acc cache key

**Comment by chaosagent** [comment]
added todo.

**Pull request merged**